src/components/ImageMapEditor/ImageMapEditor.tsx

```diff
@@ -0,0 +1,197 @@
+import {
+  ImageMapEditorContainer,
+  ImageMapEditorHeader,
+  ImageMapEditorHeaderCircle,
+  ImageMapEditorBody,
+  ImageMapEditorInput,
+  ImageMapEditorInputPlaceholder,
+  ImageMapEditorUploader,
+  ImageMapEditorImagePreview,
+  ImageMapEditorNumberTag,
+  ImageMapEditorUploaderDeleteButton,
+} from "./ImageMapEditor.style";
+
+import "react-image-crop/dist/ReactCrop.css";
+
+const CROP_BASE_COLOR = "#1070e0";
+const CROP_RESIZE_BASE_SIZE = 8;
+
+const rndBaseStyle = {
+  display: "flex",
+  padding: 5,
+  border: `solid 1px ${CROP_BASE_COLOR}`,
+  background: "transparent",
+};
+
+const RESIZE_HANDLE_BASE_STYLE = {
+  backgroundColor: CROP_BASE_COLOR,
+  width: CROP_RESIZE_BASE_SIZE,
+  height: CROP_RESIZE_BASE_SIZE,
+};
+const RESIZE_HANDLE_POSITION_OFFSET = -(CROP_RESIZE_BASE_SIZE / 2);
+
+const RESIZE_HANDLE_STYLES = {
+  bottom: {
+    ...RESIZE_HANDLE_BASE_STYLE,
+    bottom: RESIZE_HANDLE_POSITION_OFFSET,
+    left: "50%",
+    transform: "translate(-50%, 0)",
+  },
+  bottomLeft: {
+    ...RESIZE_HANDLE_BASE_STYLE,
+    left: RESIZE_HANDLE_POSITION_OFFSET,
+    bottom: RESIZE_HANDLE_POSITION_OFFSET,
+  },
+  bottomRight: {
+    ...RESIZE_HANDLE_BASE_STYLE,
+    right: RESIZE_HANDLE_POSITION_OFFSET,
+    bottom: RESIZE_HANDLE_POSITION_OFFSET,
+  },
+  left: {
+    ...RESIZE_HANDLE_BASE_STYLE,
+    left: RESIZE_HANDLE_POSITION_OFFSET,
+    top: "50%",
+    transform: "translate(0, -50%)",
+  },
+  right: {
+    ...RESIZE_HANDLE_BASE_STYLE,
+    right: RESIZE_HANDLE_POSITION_OFFSET,
+    top: "50%",
+    transform: "translate(0, -50%)",
+  },
+  top: {
+    ...RESIZE_HANDLE_BASE_STYLE,
+    top: RESIZE_HANDLE_POSITION_OFFSET,
+    left: "50%",
+    transform: "translate(-50%, 0)",
+  },
+  topLeft: {
+    ...RESIZE_HANDLE_BASE_STYLE,
+    left: RESIZE_HANDLE_POSITION_OFFSET,
+    top: RESIZE_HANDLE_POSITION_OFFSET,
+  },
+  topRight: {
+    ...RESIZE_HANDLE_BASE_STYLE,
+    right: RESIZE_HANDLE_POSITION_OFFSET,
+    top: RESIZE_HANDLE_POSITION_OFFSET,
+  },
+};
+
+interface ImageMapEditorProps {
+  coordinates: Coordinate[];
+  onDragStop: (coordinate: Coordinate) => void;
+  onResizeStop: (coordinate: Coordinate) => void;
+  onRemoveCoordinate: (coordinate: Coordinate) => void;
+  onCropComplete: (coordinate: Omit<Coordinate, "id">) => void;
+}
+
+const ImageMapEditor = ({
+  coordinates,
+  onDragStop,
+  onResizeStop,
+  onRemoveCoordinate,
+  onCropComplete,
+}: ImageMapEditorProps) => {
+  const [crop, setCrop] = useState<Crop>();
+  const [imageSrc, setImageSrc] = useState("");
+
+  const handleImageChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.files && event.target.files.length > 0) {
+      const reader = new FileReader();
+      reader.addEventListener("load", () =>
+        setImageSrc((reader?.result as string) ?? "")
+      );
+      const file = event.target.files[0];
+      if (file.type.includes("image")) {
+        reader.readAsDataURL(file);
+      }
+    }
```

Comment:
Why not early return here like everywhere else?

Reply:
you’re right, it can be early return here, more readable.